Core/Src/system.c

```diff
@@ -79,22 +96,53 @@ void system_main(System *sys) {
 
 void commandHandler(System *sys) {
 	uint8_t command;
-	uint8_t status_error = pop_ring_buffer(sys->recieve_ring_buffer, &command);
+	uint8_t status_error = pop_ring_buffer(sys->receive_ring_buffer, &command);
 	if (!status_error) {
-		switch (command) {
-		case LED1_5Hz:
-			sys->LED1_timer = sys->h_timer1;
-			break;
-		case LED1_1Hz:
-			sys->LED1_timer = sys->h_timer2;
-			break;
-		case LED2_5Hz:
-			sys->LED2_timer = sys->h_timer1;
-			break;
-		case LED2_1Hz:
-			sys->LED2_timer = sys->h_timer2;
-			break;
+
+		if (command == sys->LED1_State) {
```

Comment:
Clever usage of logic here.  Took me a second to figure out but it is a good way to catch all logic combinations of whether it is already active on the command asked for